core/invoker/src/main/scala/whisk/core/invoker/Invoker.scala

```diff
@@ -104,8 +107,24 @@ object Invoker {
       abort("Bad configuration, cannot start.")
     }
 
-    val proposedInvokerId: Option[Int] = args.headOption.map(_.toInt)
-    val assignedInvokerId = proposedInvokerId
+    // process command line arguments
+    // We accept the command line grammar of:
+    // Usage: invoker [options] [<proposedInvokerId>]
+    //    --name <value>   a unique name to use for this invoker
```

Comment:
are there any restrictions on what characters can be passed here for --name? (I assume not, like dots and dashes should work?)

Reply:
We're going to use invokerName as a piece of a zookeeper path.  So there are some modest restrictions https://zookeeper.apache.org/doc/r3.1.2/zookeeperProgrammers.html.  But I believe dots in the middle are fine (eg ip addresses or hostnames).  I didn't try it though.